### ISSUE TITLE
[BUG] - FieldType Select with SQL failing

### ISSUE THREAD

**Issue by Rzial** [bolt/bolt#7019]
Hi,
I'm getting an Exception on the Bolt backoffice related to the FieldType Select. I upgraded this project from 3.3.3 and i didn't had this problem. Tryed to recreate the database from zero but same problem popped up.

Details
-------

 - Relevant Bolt Version: 3.3.4
 - Install type: Composer install
 - PHP version: 7.0
 - DBMS: MySQL version 5.5.57
 - Used web server: Nginx version 1.12.1

Reproduction
------------
### Steps to reproduce
1. Having an entity with Select FieldType over another entity (e.g.)

```
mailing_users:
    name: Users
    singular_name: User
    fields:
        name:
            type: text
            group: "Info"
        email:
            type: text
            pattern: email
            required: true

    viewless: true
    show_on_dashboard: false
    show_in_menu: "Mailing"
    default_status: published

mailing_groups:
    name: Groups
    singular_name: Group
    fields:
        name:
            type: text
            group: "Info"
        users:
            type: select
            values: mailing_users/name,email
            keys: id
            multiple: true

    viewless: true
    show_on_dashboard: false
    show_in_menu: "Mailing"
    default_status: published
```

2. Try to create new content on `mailing_groups`

### Expected result
Be able to create/edit a new ContentType instance.

### Actual result
`SyntaxErrorException` in `AbstractMySQLDriver.php` line 90: 
An exception occurred while executing `'SELECT _mailing_users.*, GROUP_CONCAT(incomingrelation.id) as _incomingrelation_id, GROUP_CONCAT(incomingrelation.from_id) as _incomingrelation_fromid, GROUP_CONCAT(incomingrelation.from_contenttype) as _incomingrelation_fromcontenttype FROM bolt_mailing_users _mailing_users LEFT JOIN bolt_relations incomingrelation ON _mailing_users.id = incomingrelation.to_id AND incomingrelation.to_contenttype='mailing_users' GROUP BY _mailing_users.id ORDER BY ASC'`:

SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'ASC' at line 1

**Comment by bobdenotter** [comment]
We've had quirks with contenttypes with underscores before. Maybe we have a small regression. Could you try this, to make it more consistent? 

```
mailingusers:
    name: Users
    slug: mailingusers
    singular_name: User
    singular_slug: mailinguser
    fields:
        name:
            type: text
            group: "Info"
        email:
            type: text
            pattern: email
            required: true

    viewless: true
    show_on_dashboard: false
    show_in_menu: "Mailing"
    default_status: published

mailinggroups:
    name: Groups
    slug: mailinggroups
    singular_name: Group
    singular_slug: mailinggroup
    fields:
        name:
            type: text
            group: "Info"
        users:
            type: select
            values: mailingusers/name,email
            keys: id
            multiple: true

    viewless: true
    show_on_dashboard: false
    show_in_menu: "Mailing"
    default_status: published
```

**Comment by Rzial** [comment]
Just tryed, but the issue persist.

**Comment by bobdenotter** [comment]
@Rzial I'm having trouble reproducing this.. See screenshot: 

![screen shot 2017-09-24 at 09 46 46](https://user-images.githubusercontent.com/1833361/30780538-468e7596-a10d-11e7-86d0-455f70ee648d.png)

Did you remember to check/update the DB before trying?

**Comment by bobdenotter** [comment]
Scratch that, I can reproduce too. Definitely a regression in 3.3.4.. We'll look into it.

**Comment by bobdenotter** [comment]
@Rzial A fix is in the queue at #7020. Thanks for bringing this to our attention!

**Comment by eduardomart** [comment]
I am seeing this in Bolt 3.5.5, but when trying to use the search function in the front end:

![image](https://user-images.githubusercontent.com/1833020/43359524-5a5bd57c-9258-11e8-9ed2-6a22e077a754.png)
